pkg/controllers/resources/nodes/translate.go

```diff
@@ -100,6 +100,8 @@ func (s *nodeSyncer) translateUpdateBackwards(pNode *corev1.Node, vNode *corev1.
 		}
 	}
 
+	labels["kubernetes.io/hostname"] = GetNodeHost(vNode.Name)
```

Comment:
I think we should only do this if `s.useFakeKubelets` is true as otherwise we won't rewritte the hostname at all